### PR TITLE
Improving Statistics

### PR DESCRIPTION
**Pull request by ccjeremylo**
Continuation for #99 - cleaning up statistics + add tests

### CODE REVIEW

File: sporkfish/searcher/negamax.py [hunk 1]
```diff
@@ -94,7 +99,8 @@ def _negamax(
                 board, depth, capture, move, alpha
             ):
                 board.pop()
-                self._statistics.increment_pruning()
+                # add test
```
Comment (KYLChiu) [inline sporkfish/searcher/negamax.py:102]: [Future] Can go?

File: tests/test_statistics.py [hunk 2]
```diff
@@ -1,55 +1,119 @@
...
+)
+from sporkfish.searcher.searcher_config import SearcherConfig, SearchMode
+from sporkfish.searcher.searcher_factory import SearcherFactory
+from sporkfish.statistics import NodeTypes, PruningTypes, Statistics, TranpositionTable
 
 
 class TestIncrementStatistics:
-    def test_increment_node_visited(self):
+    def test_increment_visited(self):
         S = Statistics()
```
Comment (KYLChiu) [inline tests/test_statistics.py:17]: Overly extreme nitpick but capital letters are supposed to be for constants, though in a test no big deal

**Review by KYLChiu — APPROVED** [review]
Nice! Thanks for fixing the docstrings and changing to better names. Rest looks good to me, minor nitpicks, don't bother to change

File: sporkfish/searcher/negamax.py (hunk 1 above)
Comment (ccjeremylo) [reply sporkfish/searcher/negamax.py:102]: can't yet - haven't add test to check pruning

File: sporkfish/statistics.py [hunk 3]
```diff
@@ -8,124 +11,118 @@ class NodeTypes(Enum):
...
 
+class PruningTypes(Enum):
+    NULL_MOVE = "NULL_MOVE"
+    DELTA = "DELTA"
+    FUTILITY = "FUTILITY"
+    ALPHA_BETA = "ALPHA_BETA"
+
+
+class TranpositionTable(Enum):
+    TRANSPOSITITON_TABLE = "TRANSPOSITITON_TABLE"
```
[the remark this answers is not in the document]
Comment (ccjeremylo) [reply sporkfish/statistics.py:22]: happy to change in next PR

**Review by KYLChiu — APPROVED** [review]

**Pull request merged**